### PR TITLE
SAR-4489 New Interrupt Controller

### PR DESCRIPTION
**Pull request by JSimmons072019**

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/controllers/CheckCompatibleSoftwareController.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.config.AppConfig
+import uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.views.html.check_compatible_software
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.Future
+
+@Singleton
+class CheckCompatibleSoftwareController @Inject()(
```
Comment (r-melvin) [inline app/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/controllers/CheckCompatibleSoftwareController.scala]: formatting

File: conf/messages [hunk 2]
```diff
@@ -21,3 +23,10 @@ get_software.heading                                            = You will need
 get_software.line_1                                             = The software you use must be able to send income and expense reports to HMRC.
 get_software.find_software                                      = Find software that''s compatible with Making Tax Digital for Income Tax (opens in a new tab)
 get_software.continue                                           = Continue
+
+## Check for Compatible Software
+principal.check_compatible_software.title                       = Check that the software you use is compatible with Making Tax Digital for Income Tax
```
Comment (r-melvin) [inline conf/messages]: you don't need the principal prefix as this flow only has one user type

File: it/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/assets/MessageLookup.scala [hunk 3]
```diff
@@ -18,6 +18,10 @@ package uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.assets
 
 object MessageLookup {
 
+  object Base {
+    val continue = "Continue"
```
Comment (r-melvin) [inline it/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/assets/MessageLookup.scala]: Can you refactor any existing continues in the tests to use this?

File: app/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/controllers/CheckCompatibleSoftwareController.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import scala.concurrent.Future
+
+@Singleton
+class CheckCompatibleSoftwareController @Inject()(mcc: MessagesControllerComponents)
+                                                 (implicit appConfig: AppConfig) extends FrontendController(mcc) with I18nSupport {
+
+  val show: Action[AnyContent] = Action.async {
+    implicit request =>
+      Future.successful(
+        Ok(check_compatible_software(routes.TermsController.show())))
```
Comment (AlexRimmerHMRC) [inline app/uk/gov/hmrc/incometaxsubscriptioneligibilityfrontend/controllers/CheckCompatibleSoftwareController.scala]: Can you reformat this?

File: conf/app.routes [hunk 5]
```diff
@@ -7,4 +7,12 @@ GET        /overview                            uk.gov.hmrc.incometaxsubscriptio
 GET        /terms-of-participation              uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.controllers.TermsController.show
 
 ## Get Software
-GET        /get-software                        uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.controllers.GetSoftwareController.show
\ No newline at end of file
+
+GET        /get-software                        uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.controllers.GetSoftwareController.show
+
+GET        /get-software            uk.gov.hmrc.incometaxsubscriptioneligibilityfrontend.controllers.GetSoftwareController.show
```
Comment (AlexRimmerHMRC) [inline conf/app.routes]: Duplicated route, please remove

**Pull request merged**